datafusion/physical-expr/src/aggregate/tdigest.rs

```diff
@@ -658,7 +608,7 @@ impl TDigest {
 
         let max = cast_scalar_f64!(&state[3]);
         let min = cast_scalar_f64!(&state[4]);
-        assert!(max >= min);
+        assert!(max.is_nan() || min.is_nan() || max >= min);
```

Comment:
Why this change?

Reply:
max/min can be none, direct f64 comparison works not correctly in this case. replaced with `assert(max.total_cmp(&min).is_ge());`